pkg/ocm/placementbinding_test.go

```diff
@@ -0,0 +1,199 @@
+package ocm
+
+import (
+	"testing"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/stretchr/testify/assert"
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+)
+
+var (
+	defaultPlacementBindingName   = "placementbinding-test"
+	defaultPlacementBindingNsName = "test-ns"
+	defaultPlacementBindingRef    = policiesv1.PlacementSubject{
+		Name:     "placementrule-test",
+		APIGroup: "apps.open-cluster-management.io",
+		Kind:     "PlacementRule",
+	}
+	defaultPlacementBindingSubject = policiesv1.Subject{
+		Name:     "policyset-test",
+		APIGroup: "policy.open-cluster-management.io",
+		Kind:     "PolicySet",
+	}
+)
+
+func TestNewPlacementBindingBuilder(t *testing.T) {
+	// No test cases for invalid refs or subjects, those have their own unit tests.
+	testCases := []struct {
+		placementBindingName      string
+		placementBindingNamespace string
+		placementBindingRef       policiesv1.PlacementSubject
```

Comment:
Can you add unit tests for:
placementBindingRef = nil
placementBindingSubject = nil ?

Reply:
Since the structs are passed directly and not as pointers, they won't be nil. Now that you mentioned it though, the validation functions need tests for nil pointers

Reply:
I decided instead to make those functions accept a struct instead of a pointer since they're not modifying the object and then nil pointers won't be an issue